extensions/circuit/src/extension.rs

```diff
@@ -679,13 +680,17 @@ mod phantom {
             _: PhantomDiscriminant,
             a: F,
             b: F,
-            _: u16,
+            mem_start_imm: u16,
```

Comment:
This is required for wasm

Reply:
I think for this PR this is fine, but we may need to pass this pointer differently for programs that use more "fixed" memory for tables/globals (maybe add a TODO here?)